config/mcp-system/tool-call-auth.yaml

```diff
@@ -9,38 +9,59 @@ spec:
     kind: Gateway
     name: mcp-gateway
     sectionName: mcps
-  defaults:
-    rules:
-      metadata:
-        rbac: #fetches hard coded acl currrently
-          http:
-            urlExpression: '"http://acl-config.mcp-system.svc.cluster.local:8181/config/"+request.headers[":authority"]'
-      authentication: #validates the token
-        'keycloak':
-          jwt:
-            issuerUrl: http://keycloak.keycloak.svc.cluster.local/realms/mcp
-      authorization: #authorizes the tool call
-        'allow-tool-call':
-          patternMatching:
-            patterns:
-              - predicate: auth.identity.groups.exists(g, request.headers['x-mcp-toolname'] in auth.metadata.rbac.access[g])
-      response: #duplicated content may be able to solve
-        unauthorized:
-          code: 403
-          body:
-            value: |
-              {
-                "error": "Forbidden", 
-                "message": "MCP Tool Access denied. Insufficient permissions for this tool."
-              }
-        unauthenticated:
-          code: 401
+  rules:
```

Comment:
does it make sense to have multiple examples here? Perhaps for some users token exchange is a step they don't want to take yet so having a stepped approach to show examples of different options might be useful?

Reply:
Let me know if https://github.com/kagenti/mcp-gateway/pull/224/commits/f3ee5681a8ff56090f1b7b471fad7e5f0d8d4889 works